pyanaconda/modules/network/nm_client.py

```diff
@@ -213,7 +213,7 @@ def _update_bond_connection_from_ksdata(connection, network_data):
     :type network_data: pykickstart NetworkData
     """
     s_con = connection.get_setting_connection()
-    s_con.props.type = NMConnectionType.BOND
+    s_con.props.type = NMConnectionType.BOND.value
```

Comment:
I was a little worried that the enum might not work as expected. However, I don't think the new solution is safe. It will be too easy to forget to specify the value. I suggest to remove the enum and use the constants instead (for example, `NM_CONNECTION_TYPE_BOND`).

Reply:
Btw. you can also try to define the enum as `class NMConnectionType(str, Enum)`. Then the members of the enum are really strings, but I would still prefer to use the constants. https://docs.python.org/3/library/enum.html#others

Reply:
The main reason for using Enum was that I didn't want to have to import all the constants.

> I was a little worried that the enum might not work as expected. However, I don't think the new solution is safe. It will be too easy to forget to specify the value. I suggest to remove the enum and use the constants instead (for example, `NM_CONNECTION_TYPE_BOND`).

The main reason for using Enum was that I didn't want to import all the constants.

Reply:
> Btw. you can also try to define the enum as `class NMConnectionType(str, Enum)`. Then the members of the enum are really strings, but I would still prefer to use the constants. https://docs.python.org/3/library/enum.html#others

I was considering this as well, but assignment would still need using the .value
I think I will use just the constants.

Reply:
Added commit replacing enum with constants.